### PR TITLE
Participant reducer refactor

### PR DESCRIPTION
**Pull request by SUICIDALcow**

### CODE REVIEW

File: src/interfaces/stateItem.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
-interface StateItem {
-  readonly loading?: boolean
-  readonly request?: any
-  readonly data?: any
+interface StateItem<t> {
+  readonly loading: boolean
+  readonly request: { [key: string]: any }
+  readonly data: { [key: string]: t }
   readonly error?: any
```
Comment (robrobbins) [inline src/interfaces/stateItem.ts]: for `error` the TSC knows `Error`. Or you could use a union-type `Error|string|undefined` etc...

File: src/constants/index.ts [hunk 2]
```diff
@@ -1,7 +1,3 @@
-export const PARTICIPATE = 'participate'
-export const PARTICIPATE_ERROR = 'participate-error'
-export const RESET_PARTICIPANTS = 'reset-participants'
-
 export const APPLY = 'apply'
```
Comment (robrobbins) [inline src/constants/index.ts]: why? i c you moving them into the action creator files. seems like a recipe for eventually needing to move them back

File: src/interfaces/stateItem.ts (hunk 1 above)
Comment (SUICIDALcow) [reply src/interfaces/stateItem.ts]: The type of `StateItem.error` had not been updated in this PR, but I changed it to `Error | string` to address this feedback.

File: src/constants/index.ts (hunk 2 above)
Comment (SUICIDALcow) [reply src/constants/index.ts]: We can leave them in constants if you want, but in codebases I have worked on in the past, I have never had a need for the action type constants to be exposed outside of the action creators and the reducers (besides in specs, of course). It appears this is so far true in this codebase as well.

By moving the action types into the action creator files, I am trying to reduce the amount of constants that are actually used globally, making it a bit easier to read through the file.

**Pull request merged**